core/src/main/java/tc/oc/pgm/filters/LocationQueryModifier.java

```diff
@@ -0,0 +1,96 @@
+package tc.oc.pgm.filters;
+
+import javax.annotation.Nullable;
+import org.bukkit.Location;
+import org.bukkit.event.Event;
+import org.bukkit.util.Vector;
+import tc.oc.pgm.api.filter.Filter;
+import tc.oc.pgm.api.filter.query.LocationQuery;
+import tc.oc.pgm.filters.query.BlockQuery;
+
+public class LocationQueryModifier extends QueryModifier<LocationQuery> {
+
+  private final String[] coords;
+  private final boolean local;
+  Filter child;
+
+  public LocationQueryModifier(Filter child, String[] coords, boolean local) {
+    super(child);
+    this.coords = coords;
+    this.local = local;
+    this.child = child;
+  }
+
+  @Override
+  protected LocationQuery modifyQuery(LocationQuery query) {
+    Location location = query.getLocation();
+
+    Vector newVector;
+
+    if (local) {
+      newVector = parseLocalLocation(query.getLocation(), coords[0], coords[1], coords[2]);
+    } else {
+      newVector =
+          new Vector(
+              parseRelativeLocation(coords[0], location.getX()),
+              parseRelativeLocation(coords[1], location.getY()),
+              parseRelativeLocation(coords[2], location.getZ()));
+    }
+
+    return new BlockQueryCustomLocation(
+        query.getEvent(), newVector.toLocation(query.getMatch().getWorld()));
```

Comment:
You're losing pitch & yaw information when doing this.

Use `location.set(x,y,z)` to keep yaw/pitch, but watch out of unintended side-effects, i doubt it will be a thing since player#getLocation always returns a new location, but if other places create this location it could be keeping it modified even after leaving the child filter.

Alternatively just copy pitch & yaw alongisde world

Reply:
this is solved with a `clone()`, good catch!